### PR TITLE
Add link to controller on pod and replica set pages

### PR DESCRIPTION
**Pull request by spadgett**
Add the following links to the detail pages in the console using
controller references:

* pod -> replication controller (not part of deployment config)
* pod -> replica set
* pod -> build
* replica set -> deployment

We previously added a link to deployment configs using pod and
replication controller labels.

### CODE REVIEW

**Comment by spadgett** [comment]
Replaces #1574 

@jwforres Arguably a usability bug fix? We were already showing the links for deployment configs, but not deployments.

File: app/scripts/controllers/pod.js [hunk 1]
```diff
@@ -238,6 +239,19 @@ angular.module('openshiftConsole')
...
       setContainerVars();
       updateEnv();
+
+      // Show owner ref if owned by a replication controller, replica set, or build.
+      // Deployment configs are handled specially in the view.
+      delete $scope.controllerRef;
+      if (!$scope.dcName) {
+        var controllerReferences = OwnerReferencesService.getControllerReferences(pod);
+        $scope.controllerRef = _.find(controllerReferences, function(ref) {
+          return ref.kind === 'ReplicationController' ||
```
Comment (jwforres) [inline app/scripts/controllers/pod.js]: what does the owner ref on the deployer pod look like?

Comment (spadgett) [inline app/scripts/controllers/pod.js]: It's not a controller ref so won't show up.

```yaml
  ownerReferences:
    - apiVersion: v1
      kind: ReplicationController
      name: hello-openshift-2
      uid: e567f54d-5541-11e7-90d1-9e0e2b8a4114
```

**Comment by jwforres** [comment]
[merge][severity:bug]

**Pull request merged**